cosmos/cache.py

```diff
@@ -96,6 +97,25 @@ def _get_latest_partial_parse(dbt_project_path: Path, cache_dir: Path) -> Path |
     return None
 
 
+def _copy(src: str, dst: str, *, follow_symlinks: bool = True) -> None:
+    """Copy a file from `src` to `dst` and preserve metadata if possible."""
+    # Handle destination as directory
+    if os.path.isdir(dst):
+        dst = os.path.join(dst, os.path.basename(src))
+    # shutil.copy includes permission copying via chmod.
+    # If the user lacks permission to run chmod, a PermissionError occurs.
+    # To avoid this, we split the operation into two steps:
+    # first, copy the file contents; then, copy metadata if feasible without raising exceptions.
+    # Step 1: Copy file contents (no metadata)
+    shutil.copyfile(src, dst, follow_symlinks=follow_symlinks)
+
+    try:
```

Comment:
@pankajastro, what do you think if we simplified and didn't try to copy the file metadata? Step 1 should be enough

Reply:
Look like we are using modification time for partial parse and copyfile does not update that ?
https://github.com/astronomer/astronomer-cosmos/issues/1008#issuecomment-2173848972

Reply:
This should be fine. This method will be invoked when it copies the most up-to-date cache to the "persistent" cache. The fact that its timestamp is newer than the original file should not be an issue. Next time Cosmos runs, it will try to use this "persistent" cache. If, for any reason, it's outdated, it should be overridden.

Reply:
thanks for the clarification @tatiana  I have applied the feedback in 226f65440e91f1c547d711564a2062c9531afe79